src/io/ctrees_utils.c

```diff
@@ -94,6 +92,10 @@ int64_t read_locations(const char *filename, const int64_t ntrees, struct locati
              numfiles_allocated*sizeof(files_fd->fd[0]), numfiles_allocated);
     files_fd->nallocated = numfiles_allocated;
     files_fd->numfiles = 0;
+
+    files_fd->numtrees_per_file = calloc(numfiles_allocated, sizeof(files_fd->numtrees_per_file[0]));
+    XASSERT( files_fd->numtrees_per_file != NULL, MALLOC_FAILURE, "Error: Could not allocate memory of %zu bytes to hold %"PRIu32" items containing 64-bit integers",
```

Comment:
This function already is handling returning of error codes.  Can this be changed from an assert to return of `EXIT_FAILURE`?